packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useMemo, useState } from 'react'
+import { FacetColumnResultValueCount } from '@sage-bionetworks/synapse-types'
+import IconSvg from '../../../IconSvg/IconSvg'
+import { Checkbox } from '../../Checkbox'
+import { FacetFilterHeader } from '../FacetFilterHeader'
+import { EnumFacetFilterOption } from './EnumFacetFilterOption'
+import EnumFacetFilterDropdown from './EnumFacetFilterDropdown'
+
+const MAX_ENUMERATION_VALUES_TO_SHOW = 5
+
+export type RenderedFacetValue = FacetColumnResultValueCount & {
+  /* Text displayed for the option, and also used to filter in search */
+  displayText: string
+}
+
+export type EnumFacetFilterUIProps = {
+  /* The title of the faceted column to be displayed */
+  facetTitle: string
+  /* List of all facet values and information associated with each value */
+  facetValues: RenderedFacetValue[]
+  /* Whether the 'All' checkbox should be selected. */
+  allIsSelected: boolean // allIsSelected cannot be derived from facetValues[*].isSelected. See PORTALS-2680
+  /* The EnumFacetFilter can be rendered as a collapsible panel or a dropdown. Default is `'Collapsible'` */
+  containerAs?: 'Collapsible' | 'Dropdown'
+  /* If `containerAs` is a dropdown, the dropdown can be either an IconButton or a SelectBox. Default is `'Icon'` */
+  dropdownType?: 'Icon' | 'SelectBox'
+  /* If true, hides the button to collapse the panel (only shown if `containerAs` is 'Collapsible'). Default is false.*/
+  hideCollapsible?: boolean
+  /* Called when an unselected value is selected */
+  onAddValueToSelection: (value: string) => void
+  /* Called when a selected value is unselected */
+  onRemoveValueFromSelection: (value: string) => void
+  /* Called when all values should be unselected (e.g. 'All' is clicked) */
+  onRemoveAllFacetSelections: () => void
+}
+
+function removeWhitespace(value: string): string {
+  return value.replace(/\s/g, '').toLowerCase()
+}
+
+export default function EnumFacetFilterUI(props: EnumFacetFilterUIProps) {
+  const {
+    allIsSelected,
+    containerAs = 'Collapsible',
+    dropdownType = 'Icon',
+    hideCollapsible = false,
+    facetValues,
+    onAddValueToSelection,
+    onRemoveValueFromSelection,
+    facetTitle,
+    onRemoveAllFacetSelections,
+  } = props
+  const [toggleShowAll, setToggleShowAll] = useState<boolean>(false)
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(false)
+  const [showSearch, setShowSearch] = useState<boolean>(false)
+  const [searchTerm, setSearchText] = useState<string>('')
+
+  // e.g. we show a subset of facet values before hiding the rest behind "Show More"
+  // If the user has selected a value that would normally be hidden, always show all of the values
+  const valueIndexedGreaterThanMaxToShowIsSelected =
+    facetValues.length > MAX_ENUMERATION_VALUES_TO_SHOW &&
+    facetValues
+      .slice(MAX_ENUMERATION_VALUES_TO_SHOW)
+      .some(({ isSelected }) => isSelected)
```

Comment:
This is kind of strange, but it was the existing behavior. I suppose it makes sense to show you the full list if someone links you to a query where the 10th facet value is selected